dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/ValidationUtils.java

```diff
@@ -245,18 +250,82 @@ public static <T extends ValueTypedDimensionalItemObject> void validateOptionSet
+  public static void validateValueType(
+      @Nonnull Reporter reporter,
+      @Nonnull TrackerBundle bundle,
+      @Nonnull TrackerDto dto,
+      @Nonnull String value,
+      @Nonnull ValueTypedDimensionalItemObject object) {
+    ValueType valueType = object.getValueType();
+    String error;
+    if (valueIsValid(value, valueType) != null) {
+      error = String.format("Value type is %s but the value `%s` is not.", valueType, value);
+    } else {
+      error =
+          switch (valueType) {
+            case FILE_RESOURCE ->
+                bundle.getPreheat().get(FileResource.class, value) == null
+                    ? "File resource `" + value + "` could not be found."
+                    : null;
+            case IMAGE ->
+                bundle.getPreheat().get(FileResource.class, value) == null
+                    ? "File resource `" + value + "` could not be found."
+                    : validateImage(bundle, value);
+            case ORGANISATION_UNIT ->
+                bundle.getPreheat().getOrganisationUnit(value) == null
+                    ? "Organisation unit `" + value + "` could not be found."
+                    : null;
+            case USERNAME ->
+                bundle.getPreheat().getUserByUsername(value).isEmpty()
+                    ? "Username `" + value + "` could not be found."
+                    : null;
+            default -> null;
+          };
+    }
+
+    if (error != null) {
+      if (object instanceof TrackedEntityAttribute) {
+        reporter.addError(dto, ValidationCode.E1007, valueType, error);
+      } else if (object instanceof DataElement) {
+        reporter.addError(dto, ValidationCode.E1302, valueType, error);
+      }
+    }
+
+    if (valueType.isFile()) {
+      validateFileNotAlreadyAssigned(reporter, bundle, dto, value);
+    }
+  }
+
+  private static String validateImage(TrackerBundle bundle, String value) {
+    FileResource fileResource = bundle.getPreheat().get(FileResource.class, value);
+
+    return Constant.VALID_IMAGE_FORMATS.contains(fileResource.getFormat())
+        ? "File resource with uid '" + value + "' is not a valid image"
+        : null;
+  }
+
+  private static void validateFileNotAlreadyAssigned(
+      Reporter reporter,
+      TrackerBundle bundle,
+      org.hisp.dhis.tracker.imports.domain.TrackerDto trackerDto,
+      String value) {
+
+    FileResource fileResource = bundle.getPreheat().get(FileResource.class, value);
+
+    if (bundle.getStrategy(trackerDto).isCreate()) {
+      reporter.addErrorIf(
+          () -> fileResource != null && fileResource.isAssigned(), trackerDto, E1009, value);
+    }
+
+    if (bundle.getStrategy(trackerDto).isUpdate()) {
+      reporter.addErrorIf(
+          () ->
+              fileResource != null
+                  && fileResource.getFileResourceOwner() != null
+                  && !fileResource.getFileResourceOwner().equals(trackerDto.getUid().getValue()),
```

Comment:
Do we do this kind of comparison often? Not sure if its worth adding a `boolean UID.isEqualTo(@CheckForNull UidObject uid)`

Reply:
I think it would make sense but in this specific case we are not comparing on the uid of the object but on another field.